Neos.Flow/Classes/ObjectManagement/Proxy/ObjectSerializationTrait.php

```diff
@@ -131,7 +131,6 @@ private function Flow_searchForEntitiesAndStoreIdentifierArray($path, $propertyV
                 'identifier' => $identifier,
                 'entityPath' => $path
             ];
-            $this->$originalPropertyName = Arrays::setValueByPath($this->$originalPropertyName, $path, null);
```

Comment:
TBH I'm super unsure about the (side) effects of removing this line. I do see the test change below covers a sensible case though...

Reply:
/cc @bwaidelich @kdambekalns does one of you have a gut-feeling about this, or even an informed opinion?

Reply:
Argh! Sorry, I did promise to provide a rebased patch. But building one, I testet, if there would be differences. And yes there are.
Arrays containing Entities on some level would not get serialized correctly any longer, if my observation is correct. But I did not yet find the time to find a solution I would acceppt myself.

Reply:
My next try would be to create some new array structure, referencing the array and a property path to temporarily save the identifiers and use this information during wakeup. This would allow the object not to be "changed" (but to extend) during serialization and at least allows that access after serialization is possible, without breaking, what we already have.

Reply:
Thanks for the clarification :)

Reply:
I _think_ the unset was in place to make sure persisting worked. The actual entities need to be encoded, but keeping them means they _get serialized as well_, doubling data and potentially having bad side-effects. I can imagine this even dates back to our own persistence layer before Doctrine 2… So, it might not be needed anymore.

That being said, the whole concept of `Flow_searchForEntitiesAndStoreIdentifierArray` might be useless by now?!